common/js/src/executable-module/emscripten-in-offscreen-doc-main-script.js

```diff
@@ -0,0 +1,72 @@
+/**
+ * @license
+ * Copyright 2024 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * @fileoverview This file contains code to be directly executed in the
+ * Offscreen Document in order to load an Emscripten module.
+ *
+ * This is to be treated an internal implementation detail of
+ * offscreen-doc-emscripten-module.js and shouldn't be called by client code
+ * directly. It's expected to be compiled and packaged as the file called
+ * "emscripten-offscreen-doc.js" (see "emscripten-offscreen-doc.html").
+ */
+
+goog.provide('GoogleSmartCard.EmscriptenInOffscreenDocMain');
+
+goog.require('GoogleSmartCard.EmscriptenModule');
+goog.require('GoogleSmartCard.Logging');
+goog.require('GoogleSmartCard.OffscreenDocEmscriptenModule');
+goog.require('GoogleSmartCard.PortMessageChannel');
+goog.require('goog.asserts');
+
+goog.scope(function() {
+
+const GSC = GoogleSmartCard;
+
+// The URL parameters specify the name of the Emscripten module to be loaded.
+const url = new URL(globalThis.location.href);
+const moduleName =
+    url.searchParams.get(GSC.OffscreenDocEmscriptenModule.URL_PARAM);
```

Comment:
Starting to think I should better sanitize this parameter, to avoid possibility of XSS-style injection. I.e., even if someone tricks the user into opening `chrome-extension://khp.../emscripten-offscreen-doc.html?module_name=https://attacker.com/script` it shouldn't end up actually loading that third-party script.

Reply:
done